hw/top_earlgrey/util/vivado_hook_write_bitstream_pre.tcl

```diff
@@ -200,43 +232,92 @@ dict set memInfo otp brams $otp_brams
 dict set memInfo otp mem_type_regex $mem_type_regex
 dict set memInfo otp fake_word_width 0
 dict set memInfo otp addr_end_multiplier 16
+dict set memInfo otp schema "Processor"
+
+# The flash banks have 76-bit wide words. 64 bits are data, and 12 bits are metadata / integrity.
+set flash_info_brams [split [get_cells -hierarchical -filter " PRIMITIVE_TYPE =~ ${bram_regex} && NAME =~ *u_flash_ctrl*gen_prim_flash_banks[0]*gen_info_types[0].u_info_mem*gen_xpm.gen_split[0].*"] " "]
+dict set memInfo flash0_info0_data brams $flash_info_brams
+dict set memInfo flash0_info0_data mem_type_regex $mem_type_regex
+dict set memInfo flash0_info0_data fake_word_width 0
+dict set memInfo flash0_info0_data addr_end_multiplier 1
+dict set memInfo flash0_info0_data schema "MemoryArray"
+
+set flash_info_brams [split [get_cells -hierarchical -filter " PRIMITIVE_TYPE =~ ${bram_regex} && NAME =~ *u_flash_ctrl*gen_prim_flash_banks[0]*gen_info_types[1].u_info_mem*gen_xpm.gen_split[0].*"] " "]
+dict set memInfo flash0_info1_data brams $flash_info_brams
+dict set memInfo flash0_info1_data mem_type_regex $mem_type_regex
+dict set memInfo flash0_info1_data fake_word_width 0
+dict set memInfo flash0_info1_data addr_end_multiplier 1
+dict set memInfo flash0_info1_data schema "MemoryArray"
```

Comment:
Could this and the following code be deduplicated by using Tcl's `eval`?

Reply:
I went with `apply` instead, but yes.